dbt/include/athena/macros/materializations/snapshots/snapshot.sql

```diff
@@ -1,5 +1,322 @@
+              ELSE 'update'
+            END as dbt_change_type
+          , CAST('9999-01-01' as timestamp) AS dbt_valid_to
+          , True AS is_current_record
+        FROM source_data
+        LEFT JOIN snapshotted_data
+               ON snapshotted_data.dbt_unique_key = source_data.dbt_unique_key
+        WHERE snapshotted_data.dbt_unique_key IS NULL
+           OR (
+                snapshotted_data.dbt_unique_key IS NOT NULL
+            AND (
+                {{ strategy.row_changed }}
+            )
+        )
+    )
+    {%- if strategy.invalidate_hard_deletes -%}
+    {%- set target_columns = adapter.get_columns_in_relation(target_relation) -%}
+    , deletes AS (
+        SELECT
+        {% for column in target_columns if not column.name == 'dbt_snapshot_at' %}
+          {% if column.name == 'dbt_valid_from' %}
+            {{ current_timestamp() }} AS dbt_valid_from {%- if not loop.last -%},{%- endif -%}
+          {% elif column.name == 'dbt_change_type' %}
+            'delete' AS dbt_change_type {%- if not loop.last -%},{%- endif -%}
+          {% elif column.name == 'dbt_valid_to' %}
+            CAST('9999-01-01' as timestamp) AS dbt_valid_to {%- if not loop.last -%},{%- endif -%}
+          {% elif column.name == 'is_current_record' %}
+            True AS is_current_record {%- if not loop.last -%},{%- endif -%}
+          {% else %}
+            snapshotted_data.{{ column.name }} {%- if not loop.last -%},{%- endif -%}
+          {% endif %}
+        {% endfor %}
+        FROM snapshotted_data
+        LEFT JOIN source_data
+               ON snapshotted_data.dbt_unique_key = source_data.dbt_unique_key
+        WHERE source_data.dbt_unique_key IS NULL
+    )
+    SELECT * FROM upserts
+    UNION ALL
+    SELECT * FROM deletes;
+    {% else %}
+    SELECT * FROM upserts;
+    {% endif %}
+
+{%- endmacro %}
+
+{%
+  Create a new temporary table to hold the records to be upserted or deleted
+%}
+
+{% macro athena__build_snapshot_staging_table(strategy, sql, target_relation) %}
+    {%- set tmp_identifier = target_relation.identifier ~ '__dbt_tmp' -%}
+
+    {%- set tmp_relation = api.Relation.create(identifier=tmp_identifier,
+                                                  schema=target_relation.schema,
+                                                  database=none,
+                                                  type='table') -%}
+
+    {% do adapter.drop_relation(tmp_relation) %}
+
+    {%- set select = snapshot_staging_table(strategy, sql, target_relation) -%}
+
+    {% call statement('build_snapshot_staging_relation') %}
+        {{ create_table_as(False, tmp_relation, select) }}
+    {% endcall %}
+
+    {% do return(tmp_relation) %}
+{% endmacro %}
+
+{#
+    Add new columns to the table if applicable
+#}
+
+{% macro create_columns(relation, columns) -%}
+  {{ adapter.dispatch('create_columns', 'dbt')(relation, columns) }}
+{% endmacro %}
+
+{% macro default__create_columns(relation, columns) -%}
```

Comment:
here you are overwriting a default macro, should be `athena__create_columns `, as we are in the athena adapter

Reply:
I overwrote the default macro since I couldn't find something like this in the base adapter. But I have changed it to `athena__create_columns` and removed the dispatch.